Scripts/Core/EditorVR.Rays.cs

```diff
@@ -441,22 +461,52 @@ internal void PreProcessRaycastSource(Transform rayOrigin)
 
 			static Node? RequestNodeFromRayOrigin(Transform rayOrigin)
 			{
-				Node? node = null;
-				var evrDeviceData = evr.m_DeviceData;
-				for (var i = 0; i < evrDeviceData.Count; i++)
+				foreach (var deviceData in evr.m_DeviceData)
 				{
-					var deviceData = evrDeviceData[i];
 					if (!deviceData.proxy.active)
 						continue;
 
 					if (deviceData.rayOrigin == rayOrigin)
+						return deviceData.node;
+				}
+
+				foreach (var kvp in evr.GetNestedModule<MiniWorlds>().rays)
```

Comment:
It won't, it will evaluate the function once, giving you the array, and then call the rays method on that.